exporter/elasticsearchexporter/internal/objmodel/objmodel.go

```diff
@@ -81,12 +92,22 @@ const (
 	KindObject
 	KindTimestamp
 	KindIgnore
+	KindProcessor
 )
 
+// processor defines an interface to allow adding in complex structures
+// like OTLP maps and spans. Any processor type value must be processed
+// into simpler value types and added to the document via the Process
+// method. The process method is invoked by Add and AddMultiple.
+type processor interface {
+	Len() int
+	Process(*Document, string)
+}
+
```

Comment:
[**For reviewers**] Mostly repeating the same thing as what is in the issue description. The introduction of the processor is mainly for readability and in practice this may be a bit worse for the performance but I added for better testability and readability of the logic to convert complex otel structs to key value pairs. Happy to hear feedbacks or discuss this further if it looks unacceptable/too-bad.